pyiceberg/catalog/dynamodb.py

```diff
@@ -80,7 +80,15 @@
 class DynamoDbCatalog(Catalog):
     def __init__(self, name: str, **properties: str):
         super().__init__(name, **properties)
-        self.dynamodb = boto3.client(DYNAMODB_CLIENT)
+        session = boto3.Session(
+            profile_name=properties.get("profile_name"),
+            region_name=properties.get("region_name"),
+            botocore_session=properties.get("botocore_session"),
+            aws_access_key_id=properties.get("aws_access_key_id"),
+            aws_secret_access_key=properties.get("aws_secret_access_key"),
+            aws_session_token=properties.get("aws_session_token"),
+        )
+        self.dynamodb = session.client(DYNAMODB_CLIENT)
```

Comment:
Shall we add a unit test for this update? Like this one in GlueCatalog's test:
https://github.com/apache/iceberg-python/blob/d8bc9a9909f77532681344762ae71713191c5cf9/tests/catalog/test_glue.py#L442-L459

Reply:
Hey @HonahX , thanks for the suggestion. I added the unit test as you requested.

Reply:
Thanks for adding the test!